erizo_controller/erizoJS/models/Connection.js

```diff
@@ -52,27 +67,91 @@ class Connection {
       global.config.erizo.turnpass,
       global.config.erizo.networkinterface);
 
+    if (this.metadata) {
+        wrtc.setMetadata(JSON.stringify(this.metadata));
+    }
     return wrtc;
   }
 
-  _createMediaStream (id) {
+  _createMediaStream (id, mediaStreamOptions) {
     log.debug(`message: _createMediaStream, connectionId: ${this.id}, mediaStreamId: ${id}`);
     const mediaStream = new addon.MediaStream(this.threadPool, this.wrtc, id,
       this._getMediaConfiguration(this.mediaConfiguration));
     mediaStream.id = id;
+    if (mediaStreamOptions) {
+      mediaStream.metadata = mediaStreamOptions;
```

Comment:
👍 However, now that we're adding metadata here we have to reconsider what we're doing with metadata at the WebRTCConnection level since we will add there the metadata from the first stream. Does it still make sense to have different metadata at each level?

Reply:
only for logging purposes, but you're right that we'll update it in the future, so we need to follow a different approach. Anyway, I'd do it in another PR.